examples/05_graphical_pipelines.ipynb

```diff
@@ -0,0 +1,5146 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "# Graphical Pipelines\n",
+    "\n",
+    "### Content of this Notebook:\n",
+    "* Understanding what are graphical pipelines\n",
+    "* Understanding the API of graphical pipelines\n",
+    "* Examples of simple pipelines and how they can be implemented with graphical pipelines.\n",
+    "* More complex graphical pipeline (Forecasting + GridSearch)\n",
+    "* Grid search with a graphical pipeline\n",
+    "\n",
+    "\n",
+    "\n",
+    "### What are Graphical Pipelines?\n",
+    "Recap sequential pipelines:\n",
+    "\n",
+    "<img src=\"img/sequential_pipeline.png\" width=750 />\n",
+    "\n",
+    "Many tasks are non-sequential. To solve this two possibilities exist:\n",
+    "1. Nesting Sequential Pipelines.\n",
+    "2. Using Graphical Pipelines.\n",
+    "\n",
+    "\n",
+    "Thus, there is the generalised graphial pipeline.\n",
+    "* Graphical means that different steps may share the same predecessor or provide their outputs to the same successor (the dataflows can branch and merge).\n",
+    "<img src=\"img/graphical_pipeline.png\" width=750 />\n",
+    "\n",
+    "\n",
+    "* Generalised means that the pipeline can be used for multiple tasks (e.g. forecasting, classification, ...).\n",
+    "\n",
+    "**Note**\n",
+    "\n",
+    "The graphical pipeline is still experimental. Thus, this graphical should not used in production. However, we would be happy to get feedback on the graphical pipeline.\n",
+    "\n",
+    "\n",
+    "\n",
+    "### Potential Use-Cases\n",
+    "There exist various potential use-case for the graphical pipeline. In the following, we focus on a forecasting and a classification pipeline.\n",
+    "#### Forecasting Use-Case for Graphical Pipelines\n",
+    "\n",
+    "\n",
+    "The input of forecasters depends on the output of other forecasters, which same the same input.\n",
+    "* Forecaster could use the same preprocessing (branching of data flow)\n",
+    "* Forecaster could use outputs of multiple predeccessors (merging of data flow)\n",
+    "\n",
+    "<img src=\"img/graphical_pipeline_example.png\" width=900 />\n",
+    "\n",
+    "\n",
+    "**Note:** The current experimental state of the graphical pipeline does not fully support this use-case. However, we are working on this. If you are interested in this use-case and want to contribute, please contact us.\n",
```

Comment:
that's an odd note to start with.

* it appears twice - copy-paste error?
* is it still true that this cannot be done?
* irrespectively, I think it would be best to have a small paragraph at the start that explains "when to use what", and the use cases the graphical pipeline covers in addition.

Reply:
is the "more complex example" at the bottom not of this kind?

Reply:
I can imagine that even more complex stuff can be created. However, I reworked the statement